packages/hardhat-core/src/internal/core/params/argumentTypes.ts

```diff
@@ -185,14 +185,19 @@ export const inputFile: CLIArgumentType<string> = {
         throw new Error(`${strValue} is a directory, not a file`);
       }
     } catch (error) {
-      throw new HardhatError(
-        ERRORS.ARGUMENTS.INVALID_INPUT_FILE,
-        {
-          name: argName,
-          value: strValue,
-        },
-        error
-      );
+      if (error instanceof Error) {
+        throw new HardhatError(
+          ERRORS.ARGUMENTS.INVALID_INPUT_FILE,
+          {
+            name: argName,
+            value: strValue,
+          },
+          error
+        );
+      }
+
+      // eslint-disable-next-line @nomiclabs/hardhat-internal-rules/only-hardhat-error
+      throw error;
```

Comment:
Maybe we can create a helper for this pattern.

Something like

```ts
handleError(error, (e:Error) => {
   // do things here if error is an instance of Error
});
```

Reply:
I know @fvictorio has mentioned to me that our error system could use a refactor in the future; maybe we wait and add the helper into that refactor? 

Just a thought. I'm fine either way.

Reply:
Just adding my thoughts here for posterity's sake:

The original reason I handled the errors this way in this PR was because I don't know most of the areas of the code base the errors touched, so leaving it as close to current functionality was what I aimed for. In the future, either in new code or an error system refactor, we can aim to properly narrow the error type so we don't have to have eslint disables everywhere.